scripts/bootstrap/bootstrap-haskell

```diff
@@ -738,6 +738,15 @@ ask_stack() {
 	unset stack_answer
 }
 
+find_stack_root() {
+    if [ -n "${STACK_ROOT}" ] ; then
+        echo "${STACK_ROOT}"
+    elif [ -n "${STACK_XDG}" ] ; then
+        echo "${XDG_DATA_HOME}/stack"
```

Comment:
`XDG_DATA_HOME` is not always set.

The XDG specification says to assume `$HOME/.local/share` if it is not set.

Reply:
Added `$HOME/.local/share` as a default value for `$XGD_DATA_HOME` if not set.